archive/schema.go

```diff
@@ -3,49 +3,22 @@ package archive
 import (
 	"context"
 	"encoding/json"
-	"errors"
 	"os"
-	"sort"
-	"sync"
 	"time"
 
 	"github.com/brimsec/zq/pkg/iosrc"
-	"github.com/brimsec/zq/pkg/nano"
 	"github.com/brimsec/zq/zbuf"
 	"github.com/brimsec/zq/zqe"
+	"github.com/segmentio/ksuid"
 )
 
 const metadataFilename = "zar.json"
 
 type Metadata struct {
-	Version           int                  `json:"version"`
-	DataPath          string               `json:"data_path"`
-	LogSizeThreshold  int64                `json:"log_size_threshold"`
-	DataSortDirection zbuf.Direction       `json:"data_sort_direction"`
-	Spans             []SpanInfo           `json:"spans"`
-	Indexes           map[string]IndexInfo `json:"indexes"`
-}
-
-// A LogID identifies a single zng file within an archive. It is created
-// by doing a path join (with forward slashes, regardless of platform)
-// of the relative location of the file under the archive's root directory.
-type LogID string
-
-// Path returns the local filesystem path for the log file, using the
-// platforms file separator.
-func (l LogID) Path(ark *Archive) iosrc.URI {
-	return ark.DataPath.AppendPath(string(l))
-}
-
-type SpanInfo struct {
-	Span        nano.Span `json:"span"`
-	LogID       LogID     `json:"log_id"`
-	RecordCount int       `json:"record_count"`
-}
-
-type IndexInfo struct {
-	Type string `json:"type"`
-	Path string `json:"path"`
+	Version           int            `json:"version"`
+	DataPath          string         `json:"data_path"`
+	LogSizeThreshold  int64          `json:"log_size_threshold"`
+	DataSortDirection zbuf.Direction `json:"data_sort_direction"`
```

Comment:
Nit: Sort these or, if you prefer to have Version first, sort the subsequent fields (i.e., Version, DataPath, DataSortDirection, LogSizeThreshold).